lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -37,6 +72,14 @@ bool extendsFromUiPropsOrUiState(ClassDeclaration classNode) =>
         .toSource()
         .contains(RegExp('(UiProps)|(UiState)'));
 
+/// A simple RegExp against the parent of the class to verify it is `UiProps`
+/// or `UiState`.
+bool implementsUiPropsOrUiState(ClassDeclaration classNode) {
+  return classNode.implementsClause.interfaces
+      .map((typeName) => typeName.toSource())
+      .any((typeStr) => typeStr.contains(RegExp('(UiProps)|(UiState)')));
```

Comment:
#nit this matches classes that contain the string `UiProps`, which isn't the described behavior of exactly implementing `UiProps`
```suggestion
      .any({'UiProps', 'UiState'}.contains);
```